File: 979-di-string-match/di-string-match.py
class Solution:
    def diStringMatch(self, s: str) -> list[int]:
        low, high = 0, len(s)
        res = []
        
        for ch in s:
            if ch == 'I':
                res.append(low)
                low += 1
            else:  
                res.append(high)
                high -= 1
        
        res.append(low)  
        return res

File: 979-di-string-match/di-string-match.java
class Solution {
    public int[] diStringMatch(String s) 
    {
        int n = s.length();
        int[] ans = new int[n + 1];

        int low = 0, high = n, idx = 0;

        for (int i = 0; i < n; i++) 
        {
            if (s.charAt(i) == 'I')
            { 
                ans[idx++] = low++;
            }
            else 
            {
                ans[idx++] = high--;
            }
        }
        ans[idx] = low;
        return ans;
    }
}
